### ISSUE TITLE
Support division between `float` and `int`

### ISSUE THREAD

**Issue by perlun** [perlang-org/perlang#311]
The following combination of operands are currently invalid:

```
> 12.0 / (9 * 60 + 1)  
[line 1] Operands must be numbers, not double and int
```

This should be valid, since `double` and `int` are perfectly able to be combined (all 32-bit positive and negative integer values can be exactly represented in `double` per the IEEE754 specification) => the type of the operation above should be `double`.

(Side note, but we should probably not postpone implementing a single-precision `float` type for too long, since it might involve changing quite a lot of the semantics if we do it too late. See #70 for the overall ticket about implementing the missing basic data types.)

**Comment by perlun** [comment]
(We should check the other basic operations of math as well to ensure they all work, when this is fixed.)

**Comment by perlun** [comment]
> (We should check the other basic operations of math as well to ensure they all work, when this is fixed.)

#316 made the handling of binary operators much more predictable. Now, all combinations of numeric types are *explicitly handled*. No more surprises; if something is unsupported, we have a test that verifies that it emits the expected type of error (at first, runtime errors, but as of #329 they should all be validation (compile-time) errors). Once a type combination gets supported, we always have to edit the tests => the tests will from now on **always** depict a true image of the state of what type combinations are expected to work and which ones are expected to fail.

**Comment by perlun** [comment]
Fixed in #327, closing.
